test_suite/tests/test_borrow.rs

```diff
@@ -159,6 +162,47 @@ fn test_cow() {
     }
 }
 
+#[test]
+fn test_cow_str_visitor() {
+    let de_str = BorrowedStrDeserializer::<Error>::new("borrowed");
+    let de_bytes = BorrowedBytesDeserializer::<Error>::new(b"borrowed");
+
+    // This example shows, that without CowStrVisitor the result is different
+    match Cow::<str>::deserialize(de_str) {
```

Comment:
I'm guessing we can't use the new visitor in the Cow impl because that's generic?